arches/app/models/models.py

```diff
@@ -1548,7 +1549,12 @@ class Meta:
 def send_email_on_save(sender, instance, **kwargs):
     """Checks if a notification type needs to send an email, does so if email server exists"""
 
-    if instance.notif.notiftype is not None and instance.isread is False:
+    if (
+        instance.notif.notiftype is not None
+        and instance.isread is False
+        and instance.notif.context is not None
+        and ("email" in instance.notif.context)
```

Comment:
This could prevent the Exception block from being hit. I'd say either remove these lines or add an `if not` block that logs/raises an Exception instead.

Reply:
[e555cd2ff78c19579a81d5f35d21670ff7aefc97](https://github.com/archesproject/arches/commit/e555cd2ff78c19579a81d5f35d21670ff7aefc97)